### PR TITLE
Application for taking and showing picture initial interface

### PR DESCRIPTION
**Pull request by alexdtat**
Raw interface for android application.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,15 @@
+*.iml
+.gradle
+/local.properties
+/.idea/caches
```
Comment (TimPushkin) [inline .gitignore]: I tend to remove `.idea` folder completely, as we don't rely on it

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+}// Top-level build file where you can add configuration options common to all sub-projects/modules.
+plugins {
+    id 'com.android.application' version '7.2.1' apply false
+    id 'com.android.library' version '7.2.1' apply false
+    id 'org.jetbrains.kotlin.android' version '1.5.31' apply false
+}
+
+task clean(type: Delete) {
+    delete rootProject.buildDir
+}
```
Comment (TimPushkin) [inline build.gradle]: Let's add a trailing newline in every file that contains code (Gradle, Kotlin, Java) for consistency

File: app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    defaultConfig {
+        applicationId "com.example.featuredetectandroid"
+        minSdk 21
+        targetSdk 32
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        vectorDrawables {
+            useSupportLibrary true
```
Comment (TimPushkin) [inline app/build.gradle]: I don't think we need this as we set `minSdk` to `21`. So, the whole `vectorDrawables` block can be removed

File: app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    packagingOptions {
+        resources {
+            excludes += '/META-INF/{AL2.0,LGPL2.1}'
+        }
+    }
+}
+
+dependencies {
+
```
Comment (TimPushkin) [inline app/build.gradle]: It would look nicer without this newline, in my opinion.

File: app/build.gradle [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    implementation "androidx.compose.ui:ui-tooling-preview:$compose_version"
+    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.3.1'
+    implementation 'androidx.activity:activity-compose:1.3.1'
+    testImplementation 'junit:junit:4.13.2'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+    androidTestImplementation "androidx.compose.ui:ui-test-junit4:$compose_version"
+    debugImplementation "androidx.compose.ui:ui-tooling:$compose_version"
+    debugImplementation "androidx.compose.ui:ui-test-manifest:$compose_version"
+}
```
Comment (TimPushkin) [inline app/build.gradle]: Add a trailing newline, pls

File: app/build.gradle [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    packagingOptions {
+        resources {
+            excludes += '/META-INF/{AL2.0,LGPL2.1}'
+        }
+    }
+}
+
+dependencies {
+
+    implementation 'androidx.core:core-ktx:1.7.0'
```
Comment (TimPushkin) [inline app/build.gradle]: Android Studio tells me that this (and some other dependencies) can be updated.

Don't update dependencies with `compose_version`, though, as the newer version is `alpha` and may be too unstable.

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
+        android:theme="@style/Theme.FeatureDetectAndroid"
+        tools:targetApi="31">
+        <activity
+            android:name="me.alexdtat.featuredetectandroid.MainActivity"
+            android:exported="true"
+            android:label="@string/app_name"
```
Comment (TimPushkin) [inline app/src/main/AndroidManifest.xml]: Android Studio tells me that this is redundant

File: app/src/main/res/values/colors.xml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (TimPushkin) [inline app/src/main/res/values/colors.xml]: I don't think you actually need this file when using Jetpack Compose. One color from here is used in `themes.xml`, but its usage can be inlined.

File: app/src/main/res/values/themes.xml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="Theme.FeatureDetectAndroid" parent="android:Theme.Material.Light.NoActionBar">
+        <item name="android:statusBarColor">@color/purple_700</item>
```
Comment (TimPushkin) [inline app/src/main/res/values/themes.xml]: This is the only place where a color from `colors.xml` is used -- you can inline its value or even remove this line completely as you can control the status bar from Jetpack Compose too. So, the `style` block would look like this:

```xml
<style name="Theme.FeatureDetectAndroid" parent="android:Theme.Material.Light.NoActionBar" />
```

In fact, the whole file can be removed, but there are cases when you need to control app's theme like this, so I would recommend to keep it.

File: app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt [hunk 10]
```diff
@@ -0,0 +1,27 @@
+package me.alexdtat.featuredetectandroid
```
Comment (TimPushkin) [inline app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt]: I think, the name of the main package should correspond to `applicationId` in `gradle.build` and `package` in `AndroidManifest.xml`

**Review by TimPushkin — CHANGES_REQUESTED** [review]
Please, include the `gradle` folder to the files tracked by git. Its absence makes it harder to build the project.

File: .gitignore (hunk 1 above)
Comment (alexdtat) [reply .gitignore]: fixed

File: app/build.gradle (hunk 4 above)
Comment (alexdtat) [reply app/build.gradle]: fixed

File: app/build.gradle (hunk 5 above)
Comment (alexdtat) [reply app/build.gradle]: fixed

File: app/src/main/AndroidManifest.xml (hunk 7 above)
Comment (alexdtat) [reply app/src/main/AndroidManifest.xml]: fixed

File: app/build.gradle (hunk 6 above)
Comment (alexdtat) [reply app/build.gradle]: fixed

File: app/build.gradle (hunk 3 above)
Comment (alexdtat) [reply app/build.gradle]: fixed

File: app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt (hunk 10 above)
Comment (alexdtat) [reply app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt]: Will be `me.alexdtat.featuredetectandroid` ok for the whole app?

File: app/src/main/res/values/colors.xml (hunk 8 above)
Comment (alexdtat) [reply app/src/main/res/values/colors.xml]: fixed

File: build.gradle (hunk 2 above)
Comment (alexdtat) [reply build.gradle]: fixed

File: app/src/main/res/values/themes.xml (hunk 9 above)
Comment (alexdtat) [reply app/src/main/res/values/themes.xml]: fixed

File: .gitignore (hunk 1 above)
Comment (TimPushkin) [reply .gitignore]: It seems like you also accidentally removed `.DS_Store` from `.gitignore`

File: app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt (hunk 10 above)
Comment (TimPushkin) [reply app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt]: I would actually like something more general, something that will include all three of us, main developers. Let's go with `com.github.featuredetectandroid` as it is pretty common to use the hoster's domain

**Review by TimPushkin — CHANGES_REQUESTED** [review]
Change the package name and fix `.gitignore` as described above in the comments.

File: .gitignore (hunk 1 above)
Comment (alexdtat) [reply .gitignore]: Oh yes, actually. Thank you. Fixed.

File: app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt (hunk 10 above)
Comment (alexdtat) [reply app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt]: I am not very familliar with package naming traditions. Nothing offensive was meant. :)

Comment (alexdtat) [reply app/src/main/java/me/alexdtat/featuredetectandroid/MainActivity.kt]: fixed

**Review by TimPushkin — APPROVED** [review]

**Pull request merged**